Changes after test on robot - PR

DaviFSilva opened the pull request:
CHANGES:

1.  Put Telemetry on execute() method;
2.  Fix arcadeDrive clamp signal error;
3.  Fiz kF in PID equation;
4.  Add status data to telemetry on subsystems;
5.  Fix ticks per revolution of HD Hex Motor;
6.  Add speed controller to drivetrain.

DaviFSilva committed:
feat: add limiter control to drivetrain

DaviFSilva committed:
:white_check_mark: feat(tests): change code after tests in robot

gabrielatomaz commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMode.java:
```diff
@@ -48,8 +48,8 @@ public void start() {
-        subsystemsDriver.forEach(subsystem -> subsystem.execute(driver));
-        subsystemsOperator.forEach(subsystem -> subsystem.execute(operator));
+        subsystemsDriver.forEach(subsystem -> subsystem.execute(driver, telemetry));
+        subsystemsOperator.forEach(subsystem -> subsystem.execute(operator, telemetry));
```
add telemetry on init or start and use it as global on the subsystem

gabrielatomaz commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/Drivetrain.java:
```diff
@@ -55,9 +79,21 @@ public void stop() {
+
+    public void arcadeDrive(double xSpeed, double zRotation, double slowMode, double fastMode) {
+        double limiter = 0.8;
+        if (slowMode > triggerDeadBand &&  fastMode < triggerDeadBand){
```
this button logic should be done using the button listener

gabrielatomaz commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/constants/DrivetrainConstants.java:
```diff
@@ -4,4 +4,6 @@ public class DrivetrainConstants {
     public static final String MOTOR_RIGHT = "Rmotor";
     public static final String MOTOR_LEFT = "Lmotor";
     public static final String IMU = "imu";
+    public static final double MOTORS_REDUCTION = (float) 1/4 * 1/5;
```
minor:
in order to separate the constants without creating a java file for each, add a subsystem constant class inside of the global constants. for instance:
public final class GlobalConstants {

  public static final class Drivetrain {
     // add here drivetrain constants
   }
}

DaviFSilva replied:
We think that is better with different files. Because if we have many Constants classes in the same file, it would be difficult to find the classes that we want. So we will not change it.

DaviFSilva committed:
:sparkles: :bug: feat(X-drive): add XDrivetrain subsystem

Add new subsystem for Holonomic X-drive. Also, remove Telemetry from
execute method and use it as a glabal variable in each subsystem

DaviFSilva committed:
:recycle: refact(drivetrain): refact speedLimiter to use ButtonListener

DaviFSilva committed:
refact(x-drive): update x-drive subsystem

DaviFSilva committed:
chore(ftcDashboard): add ftcDashboard to gradle
